test/testaudiostreamdynamicresample.c

```diff
@@ -23,25 +23,324 @@
 
 #include <stdlib.h>
 
+#include "font8x8.h"
+
 #define SLIDER_WIDTH_PERC 500.f / 600.f
-#define SLIDER_HEIGHT_PERC 100.f / 480.f
+#define SLIDER_HEIGHT_PERC 70.f / 480.f
 
 static int done;
 static SDLTest_CommonState *state;
 
-static int multiplier = 100;
-static SDL_FRect slider_area;
-static SDL_FRect slider_fill_area;
 static SDL_AudioSpec spec;
 static SDL_AudioStream *stream;
 static Uint8 *audio_buf = NULL;
 static Uint32 audio_len = 0;
 
-static void loop(void)
+static SDL_bool auto_loop = SDL_TRUE;
+static SDL_bool auto_flush = SDL_TRUE;
+
+static Uint64 last_get_callback = 0;
+static int last_get_amount = 0;
+
+typedef struct Slider
+{
+    SDL_FRect area;
+    SDL_bool changed;
+    char fmtlabel[64];
+    float pos;
+    int type;
+    float min;
+    float mid;
+    float max;
+    float value;
+} Slider;
+
+#define NUM_SLIDERS 3
+Slider sliders[NUM_SLIDERS];
+static int active_slider = -1;
+
+static void init_slider(int index, const char* fmtlabel, int type, float value, float min, float max)
+{
+    Slider* slider = &sliders[index];
+
+    slider->area.x = state->window_w * (1.f - SLIDER_WIDTH_PERC) / 2;
+    slider->area.y = state->window_h * (0.2f + (index * SLIDER_HEIGHT_PERC * 1.4f));
+    slider->area.w = SLIDER_WIDTH_PERC * state->window_w;
+    slider->area.h = SLIDER_HEIGHT_PERC * state->window_h;
+    slider->changed = SDL_TRUE;
+    SDL_strlcpy(slider->fmtlabel, fmtlabel, SDL_arraysize(slider->fmtlabel));
+    slider->type = type;
+    slider->min = min;
+    slider->max = max;
+    slider->value = value;
+
+    if (slider->type == 0) {
+        slider->pos = (value - slider->min + 0.5f) / (slider->max - slider->min + 1.0f);
+    } else {
+        slider->pos = 0.5f;
+        slider->mid = value;
+    }
+}
+
+static float lerp(float v0, float v1, float t)
+{
+    return (1 - t) * v0 + t * v1;
+}
+
+static SDL_Texture** font_textures = NULL;
+static const int font_width = 16;
+static const int font_height = 16;
+
+static SDL_Vertex* text_vertices = NULL;
+static int* text_indices = NULL;
+static int text_reserved = 0;
+
+static void init_text_renderer()
```

Comment:
The [SDL_test library has functions to draw text](https://github.com/libsdl-org/SDL/blob/main/include/SDL3/SDL_test_font.h).

Reply:
Heh, not sure how I missed that.